src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -473,29 +473,92 @@ missing_codecs_supported (const char **codecs)
 	return TRUE;
 }
 
+typedef struct _CodecMissingSuggestedLaunchAction {
+	const gchar *message_text;
+	const gchar *button_text;
+	GDesktopAppInfo *desktop_info;
+	const gchar *mrl;
+} CodecMissingSuggestedLaunchAction;
+
+static CodecMissingSuggestedLaunchAction *
+codec_missing_suggested_launch_action_new (const gchar *message_text,
+					   const gchar *button_text,
+					   GDesktopAppInfo *desktop_info,
+					   const gchar *mrl)
+{
+	CodecMissingSuggestedLaunchAction *action = g_slice_new0 (CodecMissingSuggestedLaunchAction);
+	action->message_text = message_text;
+	action->button_text = button_text;
+	action->desktop_info = desktop_info;
+	action->mrl = mrl;
```

Comment:
We could. As it turns out, `BaconVideoWidget` will outlive `CodedMissingSuggestedLaunchAction` and the other strings live in the `.data` section of the file since they are all const. That said, being defensive doesn't hurt.